addons/project_purchase/models/project_project.py

```diff
@@ -77,7 +77,8 @@ def action_profitability_items(self, section_name, domain=None, res_id=False):
                 },
             }
             if res_id:
-                action['res_id'] = res_id
+                action['res_model'] = 'purchase.order'
+                action['res_id'] = self.env['purchase.order.line'].browse(res_id).order_id.id
```

Comment:
```suggestion
                action['res_id'] = self.env['purchase.order'].search([('order_id', '=', res_id)]).id
```
To reduce the query, we need to search the order directly instead of the order line. Another thing, is the action's title `Purchase Order Items` okay, given that we are opening the purchase order form view.